src/main/java/org/geonetwork/index/model/record/IndexRecord.java

```diff
@@ -704,7 +714,9 @@ private void handleOperationsProperties(String name, Object value) {
     ArrayList<Integer> operationField =
         operations.computeIfAbsent(name, k -> new ArrayList<Integer>());
     if (value instanceof List) {
-      operationField.addAll((Collection) value);
+      @SuppressWarnings("unchecked")
+      List<Integer> list = (List<Integer>) value;
+      operationField.addAll(list);
```

Comment:
Introduce local variables to address unchecked warnings with relatively minimal impact; and perhaps a slight improvement to readability. This is a real limitation of the Java language...